geonetwork/content.md

```diff
@@ -12,50 +12,130 @@ The project is part of the Open Source Geospatial Foundation ( [OSGeo](http://ww
 
 %%LOGO%%
 
-# How to use this image
+## How to use this image
+
+GeoNetwork 4 uses an Elasticsearch server to store the index of the documents it manages so **it can't be run without configuring the URL of the Elasticsearch server**.
+
```

Comment:
```suggestion

To use this image:

1. Use docker pull to download Elasticsearch compatible with GeoNetwork:
   
    > docker pull docker.elastic.co/elasticsearch/elasticsearch:7.17.15

2. Use docker run, leaving port 9200 available for communication with GeoNetwork:
   
   > docker run -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:7.17.15

3. Use docker pull to download GeoNetwork stable release:

   > docker pull geonetwork:4.2
   
4. Use docker run, leaving port 8080 for browser interaction:

   > docker run --name geonetwork-stable -d -p 8080:8080 geonetwork:4.2
   
5. Browse to http://localhost:8080/geonetwork to setup your catalogue.
```

Reply:
Because I was using the GitHub editor I use block quotes to outline step by step instructions.

Reply:
And this was wrong, when running I get:
```
docker run -e ES_HOST=elasticsearch geonetwork:4.2.8
```
```
WARNING: Environment variable ES_HOST is mandatory

GeoNetwork requires an Elasticsearch instance to store the index.
Please define the variable ES_HOST with the Elasticsearch 
host name. For example

docker run -e ES_HOST=elasticsearch geonetwork:4.2.8
```

Reply:
I found a way to get it to run, but I suspect there is an easier way:

1. Start the initial elasticsearch with the name `geonetwork-index`:
```
docker run --name geonetwork-index -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:7.17.15
```

2. Look up the IP address: 

```
docker container inspect geonetwork-index | grep -I IPAddress
```

3. When starting geonetwork use --add-host to allow ES_HOST to work:
```
docker run --name geonetwork-stable --add-host=geonetwork-index:172.17.0.2 -p 8080:8080 -e ES_HOST=geonetwork-index geonetwork:4.2
```

Reply:
It's easier if you add geonetwork and elasticsearch to the same network.
About the `ES_HOST` variable, it is required for GN 4 < 4.4.0.

Reply:
```console
docker network create gn-network
docker run --network gn-network --name gn-elasticsearch -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:7.17.15
docker run --network gn-network --name geonetwork-latest -p 8080:8080 -e GN_CONFIG_PROPERTIES="-Des.host=gn-elasticsearch -Des.protocol=http -Des.port=9200 -Des.url=http://gn-elasticsearch:9200" geonetwork:latest
```

Reply:
If that is what you recommend, then that is what this section should start out with.

Goal is to have a complete example :)